### PR TITLE
Add par & parWith to Managed

### LINKED ISSUE #257: Support par/parWith in Managed
See discussion in #252 for details

### PR DESCRIPTION
**Pull request by mijicd**
This pull request resolves #257.

### CODE REVIEW

File: core/shared/src/main/scala/scalaz/zio/Managed.scala [hunk 1]
```diff
@@ -1,6 +1,12 @@
 package scalaz.zio
 
 sealed abstract class Managed[+E, +R] extends Serializable { self =>
+  type R0 <: R
+
+  def acquire: IO[E, R0] = IO.never
```
Comment (iravid) [inline core/shared/src/main/scala/scalaz/zio/Managed.scala]: Why not keep this and `release` abstract?

Comment (mijicd) [reply core/shared/src/main/scala/scalaz/zio/Managed.scala]: @iravid thanks for reviewing, will take a look!

Comment (mijicd) [reply core/shared/src/main/scala/scalaz/zio/Managed.scala]: @iravid please re-review.

File: core/shared/src/main/scala/scalaz/zio/Managed.scala [hunk 2]
```diff
@@ -12,75 +12,130 @@ package scalaz.zio
  * some checked error, as per the type of the functions provided by the resource.
  */
 sealed abstract class Managed[+E, +R] extends Serializable { self =>
-  def use[E1 >: E, A](f: R => IO[E1, A]): IO[E1, A]
+  type R0 <: R
+
+  def acquire: IO[E, R0]
```
Comment (iravid) [inline core/shared/src/main/scala/scalaz/zio/Managed.scala]: `acquire` and `release` should be private

**Comment by mijicd** [comment]
@iravid They should be protected, since it is not possible to override private members afaik. Other than that I agree, will push the corrected version later today.

**Comment by iravid** [comment]
Yes indeed. Also, I still think you should not provide default implementations for them.

**Comment by mijicd** [comment]
@iravid All changes done. Side question: should `liftIO` be kept as it is unsafe?

**Comment by iravid** [comment]
It is definitely useful (and I use it all the time), and only unsafe if the user chooses to acquire a resource without bracketing in the first place - so I'd say that we should :-)

**Comment by jdegoes** [comment]
@iravid Thanks for the review!

**Pull request merged**